### PR TITLE
Feature syst speed

### PR DESCRIPTION
**Pull request by willp240**
This PR is designed to speed up the calculation of the LLH when using systematics. The BinnedEdShrinker is used to take out the buffer region and reassign bin numbers. Previously, the reassigning of bin numbers happened everytime the LLH was evaluated. This PR introduces a SetBinMap function which is only called the first time the shrinker is called. It then saves the corresponding bin numbers in a vector (fBinVec) to avoid having to do the nested loop of old and new bins (unshrunk and shrunk) every time.

Most of it is just moving code from BinnedEDShrinker::ShrinkDist to SetBinMap, so it looks like more has changed than it has really.

Bar any coding typos, I think the logic works unless there's some situation where the buffer would change during a fit, but maybe there's something I've not thought of?

I have been using it for a while with the double beta fits and haven't seen any problems, but it would be good to get a second pair of eyes to take a look at it.

### CODE REVIEW

**Comment by dcookman** [comment]
@willp240 Before we merge this, does all of the unit tests pass? If the unit test coverage is good enough any logic bugs should get caught.

File: src/fitutil/BinnedEDShrinker.cpp
Comment (dcookman) [inline src/fitutil/BinnedEDShrinker.cpp:106]: I'm getting one pedantic compiler warning here - we're comparing an signed integer to an unsigned one. Could you cast one of these on this line so that the signedness matches? (I'm on the war path to ensure no warnings in OXO for as long as I'm a maintainer!)

**Comment by dcookman** [comment]
@willp240 Compiles & passes for me too. Only one annoying compiler warning - if you could sort that out (it's tiny!) then I'll merge this in.

File: src/fitutil/BinnedEDShrinker.cpp
Comment (willp240) [reply src/fitutil/BinnedEDShrinker.cpp]: Good spot! Now fixed, thanks!

**Review by dcookman — APPROVED** [review]
Brill thanks for the fix.

**Pull request merged**
